tests/manage/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py

```diff
@@ -0,0 +1,372 @@
+from tests.manage.z_cluster.pvc_ops import test_create_delete_pvcs
+
+logger = logging.getLogger(__name__)
+
+# TO DO: replace/remove this with actual workloads like couchbase, amq and
+# pgsql later
+
+
+def check_nodes_status():
+    """
+    This function runs in a loop to check the status of nodes. If the node(s) are in NotReady state then an
+    exception is raised. Note: this function needs to be run as a background thread during the execution of a test
+    """
+    node.wait_for_nodes_status(
+        node_names=None,
+        status=constants.NODE_READY,
+        timeout=5)
+    logger.info("All master and worker nodes are in Ready state.")
+    return True
+
+
+@pytest.mark.parametrize(
+    argnames=["percent_to_fill"],
+    argvalues=[
+        pytest.param(
+            *[10],
+            marks=pytest.mark.polarion_id("OCS-2131")
+        ),
+    ]
+)
+@ignore_leftovers
+@tier1
+class TestAddCapacity(ManageTest):
+    def test_add_capacity(self, project_factory, multi_dc_pod, multi_pvc_factory,
+                          pod_factory, mcg_obj, awscli_pod, bucket_factory, percent_to_fill
+                          ):
+
+        #####################################
+        #           ENTRY CRITERIA          #
+        #####################################
+        # Prepare initial configuration : logger, cluster filling, loop for creating & deleting of PVCs and Pods,
+        # noobaa IOs etc.,
+
+        # Perform Health checks:
+        # Make sure cluster is healthy
+
+        assert ceph_health_check(
+            defaults.ROOK_CLUSTER_NAMESPACE), "Entry criteria FAILED: Cluster is Unhealthy"
+
+        # All OCS pods are in running state:
+        assert pod_helpers.check_pods_in_running_state(), \
+            "Entry criteria FAILED: one or more OCS pods are not in running state"
+
+        # Create the namespace under which this test will executeq:
+        project = project_factory()
+
+        # total pvc created will be 'num_of_pvcs' * 4 types of pvcs(rbd-rwo,rwx
+        # & cephfs-rwo,rwx)
+        num_of_pvcs = 50
+
+        rwo_rbd_pods = multi_dc_pod(
+            num_of_pvcs=num_of_pvcs,
+            pvc_size=175,
+            project=project,
+            access_mode="RWO",
+            pool_type='rbd',
+            timeout=360
+        )
+        # Note: Skipping cephfs pods creation
+        # observing bug https://bugzilla.redhat.com/show_bug.cgi?id=1785399,
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1779421#c14
+
+        # rwo_cephfs_pods = multi_dc_pod(
+        #     num_of_pvcs=num_of_pvcs,
+        #     pvc_size=175,
+        #     project=project,
+        #     access_mode="RWO",
+        #     pool_type='cephfs'
+        # )
+        # rwx_cephfs_pods = multi_dc_pod(
+        #     num_of_pvcs=num_of_pvcs,
+        #     pvc_size=175,
+        #     project=project,
+        #     access_mode="RWX",
+        #     pool_type='cephfs'
+        # )
+        # Create rwx-rbd pods
```

Comment:
needed?

Reply:
I think till bz gets resolved, we can remove these commented lines. Later we can add